### PR TITLE
Change UnixListener to use IoSource

### PR DESCRIPTION
**Pull request by Thomasdezeeuw**
Depends on #1245.

### CODE REVIEW

**Comment by Thomasdezeeuw** [comment]
Oh.. GitHub changed my commit message to "Merge all sys:🐚:uds files", but its `Merge all sys::shell::uds files`. Locally it uses the text.

**Comment by kleimkuhler** [comment]
I know this shouldn't really change much, but I'll give it a review when it's rebased after #1245 merges!

**Comment by Thomasdezeeuw** [comment]
@kleimkuhler I've rebased this.

File: src/sys/unix/uds/listener.rs [hunk 1]
```diff
@@ -1,148 +1,84 @@
...
+    let socket = syscall!(accept(
+        listener.as_raw_fd(),
+        &mut sockaddr as *mut libc::sockaddr_un as *mut libc::sockaddr,
+        &mut socklen,
+    ))
+    .and_then(|socket| {
+        // Ensure the socket is closed if either of the `fcntl` calls
+        // error below.
+        let s = unsafe { net::UnixStream::from_raw_fd(socket) };
+        syscall!(fcntl(socket, libc::F_SETFL, libc::O_NONBLOCK))
```
Comment (kleimkuhler) [inline src/sys/unix/uds/listener.rs]: Remove this in a separate change to reflect `tcp::accept` where we know the new FD inherits file status?

Comment (Thomasdezeeuw) [reply src/sys/unix/uds/listener.rs]: Created #1250.

**Pull request merged**